zipline/pipeline/engine.py

```diff
@@ -170,7 +179,10 @@ def run_pipeline(self, pipeline, start_date, end_date):
             graph,
             dates,
             assets,
-            initial_workspace={self._root_mask_term: root_mask_values},
+            initial_workspace={
+                self._root_mask_term: root_mask_values,
+                self._root_mask_dates_term: dates.values[:, None],
```

Comment:
I think this is what I actually wrote `as_column` for and then forgot.  Will update.